### ISSUE TITLE
pizzly fusion_caller GTF file contains 0 genes and 0 transcripts if disambiguate is on

### ISSUE THREAD

**Issue by apastore** [bcbio/bcbio-nextgen#2103]
Hi Rorky, I would like to detect fusion on some PDX model,  Pizzly GTF is empty, I guess the region is because /work/inputs/transcriptome/hg19-mm10.fa does not have a gif file.

Any suggestion on what I could do?
Thanks!
Alessandro 

ls -latr work/pizzly/
total 64
-rw-r--r--  1 pastore abdelwao     0 Oct 13 16:01 pizzly.cache
-rw-r--r--  1 pastore abdelwao     0 Oct 13 16:01 pizzly.gtf
drwxr-xr-x  2 pastore abdelwao   512 Oct 13 16:01 .
drwxr-xr-x 15 pastore abdelwao 32768 Oct 13 16:02 ..

# Template for human RNA-seq using Illumina prepared samples
---
details:
  - analysis: RNA-seq
    genome_build: hg19
    algorithm:
      aligner: star
      disambiguate: mm10
      strandedness: unstranded
      transcriptome_align: false
      fusion_mode: true
      #fusion_caller: oncofuse
upload:
  dir: ../final
resources:
  samtools:
    memory: 8g

[2017-10-13T20:01Z] gpu-2-7.local: Running pizzly on /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/CNS15PDXPCItreated/quant/fusion.txt.
[2017-10-13T20:01Z] gpu-1-6.local: Running pizzly on /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/SUDHL5IGO05667/quant/fusion.txt.
[2017-10-13T20:01Z] gpu-1-10.local: Running pizzly on /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/HBL01IGO05667/quant/fusion.txt.
[2017-10-13T20:01Z] gpu-1-17.local: Running pizzly on /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/CNS15PDXuntreated/quant/fusion.txt.
[2017-10-13T20:01Z] gpu-2-9.local: Running pizzly on /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/CNS14PDXuntreated/quant/fusion.txt.
[2017-10-13T20:01Z] gpu-2-6.local: Running pizzly on /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/TMD08IGO05667/quant/fusion.txt.
[2017-10-13T20:01Z] gpu-1-16.local: Running pizzly on /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/SUDHL6IGO05667/quant/fusion.txt.
[2017-10-13T20:01Z] gpu-1-4.local: Running pizzly on /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/CNS16IGO05667/quant/fusion.txt.
[2017-10-13T20:01Z] gpu-2-9.local: GTF file contains 0 genes and 0 transcripts
[2017-10-13T20:01Z] gpu-2-7.local: GTF file contains 0 genes and 0 transcripts
[2017-10-13T20:01Z] gpu-1-6.local: GTF file contains 0 genes and 0 transcripts
[2017-10-13T20:01Z] gpu-1-10.local: GTF file contains 0 genes and 0 transcripts
[2017-10-13T20:01Z] gpu-2-6.local: GTF file contains 0 genes and 0 transcripts
[2017-10-13T20:01Z] gpu-1-17.local: Opening cached file ... loaded 0 genes and 0 transcripts
[2017-10-13T20:01Z] gpu-1-16.local: Opening cached file ... loaded 0 genes and 0 transcripts
[2017-10-13T20:01Z] gpu-1-4.local: Opening cached file ... loaded 0 genes and 0 transcripts
[2017-10-13T20:01Z] gpu-1-14.local: Running pizzly on /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/CNS13IGO05667/quant/fusion.txt.
[2017-10-13T20:01Z] gpu-1-12.local: Running pizzly on /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/CNS17IGO05667/quant/fusion.txt.
[2017-10-13T20:01Z] gpu-1-14.local: Opening cached file ... loaded 0 genes and 0 transcripts
[2017-10-13T20:01Z] gpu-2-4.local: Running pizzly on /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/OCILY3IGO05667/quant/fusion.txt.
[2017-10-13T20:01Z] gpu-1-12.local: Opening cached file ... loaded 0 genes and 0 transcripts
[2017-10-13T20:01Z] gpu-2-10.local: Running pizzly on /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/CNS10PDXuntreated/quant/fusion.txt.
[2017-10-13T20:01Z] gpu-2-8.local: Running pizzly on /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/SUDHL10IGO05667/quant/fusion.txt.
[2017-10-13T20:01Z] gpu-1-15.local: Running pizzly on /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/CNS6PDXuntreated/quant/fusion.txt.
[2017-10-13T20:01Z] gpu-1-8.local: Running pizzly on /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/CNS11PDXuntreated/quant/fusion.txt.
[2017-10-13T20:01Z] gpu-2-4.local: Opening cached file ... loaded 0 genes and 0 transcripts
[2017-10-13T20:01Z] gpu-2-10.local: Opening cached file ... loaded 0 genes and 0 transcripts
[2017-10-13T20:01Z] gpu-1-15.local: Opening cached file ... loaded 0 genes and 0 transcripts
[2017-10-13T20:01Z] gpu-2-8.local: Opening cached file ... loaded 0 genes and 0 transcripts
[2017-10-13T20:01Z] gpu-1-8.local: Opening cached file ... loaded 0 genes and 0 transcripts
[2017-10-13T20:01Z] gpu-1-10.local: Error, could not find any transcript sequences check that the ids in the FASTA file and GTF file match
[2017-10-13T20:01Z] gpu-2-9.local: Error, could not find any transcript sequences check that the ids in the FASTA file and GTF file match
[2017-10-13T20:01Z] gpu-1-4.local: Error, could not find any transcript sequences check that the ids in the FASTA file and GTF file match
[2017-10-13T20:01Z] gpu-1-6.local: Error, could not find any transcript sequences check that the ids in the FASTA file and GTF file match
[2017-10-13T20:01Z] gpu-1-17.local: Error, could not find any transcript sequences check that the ids in the FASTA file and GTF file match
[2017-10-13T20:01Z] gpu-1-16.local: Error, could not find any transcript sequences check that the ids in the FASTA file and GTF file match
[2017-10-13T20:01Z] gpu-2-7.local: Error, could not find any transcript sequences check that the ids in the FASTA file and GTF file match
[2017-10-13T20:01Z] gpu-2-6.local: Error, could not find any transcript sequences check that the ids in the FASTA file and GTF file match
[2017-10-13T20:01Z] gpu-1-10.local: Uncaught exception occurred
Traceback (most recent call last):
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/bcbio/provenance/do.py", line 23, in run
    _do_run(cmd, checks, log_stdout, env=env)
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/bcbio/provenance/do.py", line 103, in _do_run
    raise subprocess.CalledProcessError(exitcode, error_msg)
CalledProcessError: Command 'set -o pipefail; /cbio/ski/abdelwao/nobackup/pastore/bcbio/galaxy/../anaconda/bin/pizzly -k 31 --gtf /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/pizzly/pizzly.gtf --cache /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/pizzly/pizzly.cache --align-score 2 --insert-size 177 --fasta /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/inputs/transcriptome/hg19-mm10.fa --output /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/bcbiotx/tmpBNLREl/HBL01IGO05667/HBL01IGO05667 /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/HBL01IGO05667/quant/fusion.txt
GTF file contains 0 genes and 0 transcripts
Error, could not find any transcript sequences check that the ids in the FASTA file and GTF file match
' returned non-zero exit status 1
[2017-10-13T20:01Z] gpu-1-6.local: Uncaught exception occurred
Traceback (most recent call last):
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/bcbio/provenance/do.py", line 23, in run
    _do_run(cmd, checks, log_stdout, env=env)
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/bcbio/provenance/do.py", line 103, in _do_run
    raise subprocess.CalledProcessError(exitcode, error_msg)
CalledProcessError: Command 'set -o pipefail; /cbio/ski/abdelwao/nobackup/pastore/bcbio/galaxy/../anaconda/bin/pizzly -k 31 --gtf /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/pizzly/pizzly.gtf --cache /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/pizzly/pizzly.cache --align-score 2 --insert-size 187 --fasta /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/inputs/transcriptome/hg19-mm10.fa --output /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/bcbiotx/tmpv3QjTz/SUDHL5IGO05667/SUDHL5IGO05667 /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/SUDHL5IGO05667/quant/fusion.txt
GTF file contains 0 genes and 0 transcripts
Error, could not find any transcript sequences check that the ids in the FASTA file and GTF file match
' returned non-zero exit status 1
[2017-10-13T20:01Z] gpu-1-17.local: Uncaught exception occurred
Traceback (most recent call last):
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/bcbio/provenance/do.py", line 23, in run
    _do_run(cmd, checks, log_stdout, env=env)
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/bcbio/provenance/do.py", line 103, in _do_run
    raise subprocess.CalledProcessError(exitcode, error_msg)
CalledProcessError: Command 'set -o pipefail; /cbio/ski/abdelwao/nobackup/pastore/bcbio/galaxy/../anaconda/bin/pizzly -k 31 --gtf /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/pizzly/pizzly.gtf --cache /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/pizzly/pizzly.cache --align-score 2 --insert-size 198 --fasta /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/inputs/transcriptome/hg19-mm10.fa --output /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/bcbiotx/tmpMnkkmV/CNS15PDXuntreated/CNS15PDXuntreated /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/CNS15PDXuntreated/quant/fusion.txt
Opening cached file ... loaded 0 genes and 0 transcripts

**Comment by apastore** [comment]
Hi @roryk , I have rerun pizzly using hg19 only (without disambiguation) but still I get an empty pizzly.gtfv file... any Idea what I a doing wrong?

Thanks!
Alessandro 

Opening cached file ... loaded 0 genes and 0 transcripts
Error, could not find any transcript sequences check that the ids in the FASTA file and GTF file match
' returned non-zero exit status 1
2017-10-14 00:18:30.617 [IPClusterStop] Using existing profile dir: u'/cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/log/ipython'
2017-10-14 00:18:30.658 [IPClusterStop] Stopping cluster [pid=15038] with [signal=2]
2017-10-14 00:18:31.947 [IPClusterStop] Using existing profile dir: u'/cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/log/ipython'
2017-10-14 00:18:31.956 [IPClusterStop] Stopping cluster [pid=15038] with [signal=2]
15 Engines running
Sending a shutdown signal to the controller and engines.
1 Engines running
Sending a shutdown signal to the controller and engines.
15 Engines running
Sending a shutdown signal to the controller and engines.
Traceback (most recent call last):
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/bin/bcbio_nextgen.py", line 234, in <module>
    main(**kwargs)
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/bin/bcbio_nextgen.py", line 43, in main
    run_main(**kwargs)
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/bcbio/pipeline/main.py", line 50, in run_main
    fc_dir, run_info_yaml)
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/bcbio/pipeline/main.py", line 86, in _run_toplevel
    for xs in pipeline(config, run_info_yaml, parallel, dirs, samples):
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/bcbio/pipeline/main.py", line 243, in rnaseqpipeline
    samples = rnaseq.quantitate_expression_parallel(samples, run_parallel)
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/bcbio/pipeline/rnaseq.py", line 118, in quantitate_expression_parallel
    samples = run_parallel("detect_fusions", samples)
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/ipython.py", line 136, in run
    for data in view.map_sync(fn, items, track=False):
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/ipyparallel/client/view.py", line 353, in map_sync
    return self.map(f,*sequences,**kwargs)
  File "<decorator-gen-139>", line 2, in map
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/ipyparallel/client/view.py", line 54, in sync_results
    ret = f(self, *args, **kwargs)
  File "<decorator-gen-138>", line 2, in map
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/ipyparallel/client/view.py", line 39, in save_ids
    ret = f(self, *args, **kwargs)
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/ipyparallel/client/view.py", line 1119, in map
    return pf.map(*sequences)
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/ipyparallel/client/remotefunction.py", line 268, in map
    ret = self(*sequences)
  File "<decorator-gen-118>", line 2, in __call__
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/ipyparallel/client/remotefunction.py", line 75, in sync_view_results
    return f(self, *args, **kwargs)
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/ipyparallel/client/remotefunction.py", line 251, in __call__
    return r.get()
  File "/cbio/ski/abdelwao/home/pastore/nobackup/bcbio/anaconda/lib/python2.7/site-packages/ipyparallel/client/asyncresult.py", line 104, in get
    raise self._exception
ipyparallel.error.CompositeError: one or more exceptions from call to method: detect_fusions
[6:apply]: CalledProcessError: Command 'set -o pipefail; /cbio/ski/abdelwao/nobackup/pastore/bcbio/galaxy/../anaconda/bin/pizzly -k 31 --gtf /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/pizzly/pizzly.gtf --cache /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/pizzly/pizzly.cache --align-score 2 --insert-size 191 --fasta /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/inputs/transcriptome/hg19.fa --output /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/bcbiotx/tmpIQTeK5/CNS10PDXuntreated/CNS10PDXuntreated /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/CNS10PDXuntreated/quant/fusion.txt
Opening cached file ... loaded 0 genes and 0 transcripts
Error, could not find any transcript sequences check that the ids in the FASTA file and GTF file match
' returned non-zero exit status 1
[14:apply]: CalledProcessError: Command 'set -o pipefail; /cbio/ski/abdelwao/nobackup/pastore/bcbio/galaxy/../anaconda/bin/pizzly -k 31 --gtf /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/pizzly/pizzly.gtf --cache /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/pizzly/pizzly.cache --align-score 2 --insert-size 193 --fasta /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/inputs/transcriptome/hg19.fa --output /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/bcbiotx/tmpgVZ2ro/CNS11PDXuntreated/CNS11PDXuntreated /cbio/ski/abdelwao/nobackup/pastore/install/bcbio-vm/data/project/christian/5667_AZ/5667_AZ/work/kallisto/CNS11PDXuntreated/quant/fusion.txt
Opening cached file ... loaded 0 genes and 0 transcripts

**Comment by roryk** [comment]
Hi @apastore,

Sorry about that, pizzly doesn't look like it is working right when disambiguate is on. I'll push a fix for that tomorrow. Could you delete the pizzly directory and rerun? I think what is happening is the non-disambiguate run is picking up the empty pizzly cache from the disambiguate run.

**Comment by apastore** [comment]
Thanks @roryk ,
I have delete again the pizzly folder on work, but still I get an empty gtf for hg19... any suggestion.. I have no idea. 

Thanks a lot!

total 64
-rw-r--r--  1 pastore abdelwao     0 Oct 17 20:11 pizzly.cache
-rw-r--r--  1 pastore abdelwao     0 Oct 17 20:11 pizzly.gtf
drwxr-xr-x  2 pastore abdelwao   512 Oct 17 20:11 .
drwxr-xr-x 13 pastore abdelwao 32768 Oct 17 20:11 ..

**Comment by roryk** [comment]
Hi Paul,

Thanks-- could you pass along the input configuration so I can figure out what is going wrong? Sorry about the problems.

**Comment by apastore** [comment]
Sure! thanks a lot! I can also try to completely remove the run, if you think this may be the issue 

# Template for human RNA-seq using Illumina prepared samples
---
details:
  - analysis: RNA-seq
    genome_build: hg19
    algorithm:
      aligner: star
      #disambiguate: mm10
      strandedness: unstranded
      transcriptome_align: false
      fusion_mode: true
      #fusion_caller: oncofuse
upload:
  dir: ../final
resources:
  samtools:
    memory: 8g

**Comment by apastore** [comment]
I case you need, this is the head of the hg19 gtf
$ head ref-transcripts.gtf

chr1	processed_transcript	transcript	11869	14409	.	+	.	gene_id "ENSG00000223972"; transcript_id "ENST00000456328"; gene_name "DDX11L1"; gene_source "ensembl_havana"; gene_biotype "pseudogene"; transcript_name "DDX11L1-002"; transcript_source "havana";
chr1	processed_transcript	exon	11869	12227	.	+	.	gene_id "ENSG00000223972"; transcript_id "ENST00000456328"; exon_number "1"; gene_name "DDX11L1"; gene_source "ensembl_havana"; gene_biotype "pseudogene"; transcript_name "DDX11L1-002"; transcript_source "havana"; exon_id "ENSE00002234944";
chr1	processed_transcript	exon	12613	12721	.	+	.	gene_id "ENSG00000223972"; transcript_id "ENST00000456328"; exon_number "2"; gene_name "DDX11L1"; gene_source "ensembl_havana"; gene_biotype "pseudogene"; transcript_name "DDX11L1-002"; transcript_source "havana"; exon_id "ENSE00003582793";
chr1	processed_transcript	exon	13221	14409	.	+	.	gene_id "ENSG00000223972"; transcript_id "ENST00000456328"; exon_number "3"; gene_name "DDX11L1"; gene_source "ensembl_havana"; gene_biotype "pseudogene"; transcript_name "DDX11L1-002"; transcript_source "havana"; exon_id "ENSE00002312635";
chr1	transcribed_unprocessed_pseudogene	transcript	11872	14412	.	+	.	gene_id "ENSG00000223972"; transcript_id "ENST00000515242"; gene_name "DDX11L1"; gene_source "ensembl_havana"; gene_biotype "pseudogene"; transcript_name "DDX11L1-201"; transcript_source "ensembl";
chr1	transcribed_unprocessed_pseudogene	exon	11872	12227	.	+	.	gene_id "ENSG00000223972"; transcript_id "ENST00000515242"; exon_number "1"; gene_name "DDX11L1"; gene_source "ensembl_havana"; gene_biotype "pseudogene"; transcript_name "DDX11L1-201"; transcript_source "ensembl"; exon_id "ENSE00002234632";
chr1	transcribed_unprocessed_pseudogene	exon	12613	12721	.	+	.	gene_id "ENSG00000223972"; transcript_id "ENST00000515242"; exon_number "2"; gene_name "DDX11L1"; gene_source "ensembl_havana"; gene_biotype "pseudogene"; transcript_name "DDX11L1-201"; transcript_source "ensembl"; exon_id "ENSE00003608237";
chr1	transcribed_unprocessed_pseudogene	exon	13225	14412	.	+	.	gene_id "ENSG00000223972"; transcript_id "ENST00000515242"; exon_number "3"; gene_name "DDX11L1"; gene_source "ensembl_havana"; gene_biotype "pseudogene"; transcript_name "DDX11L1-201"; transcript_source "ensembl"; exon_id "ENSE00002306041";
chr1	transcribed_unprocessed_pseudogene	transcript	11874	14409	.	+	.	gene_id "ENSG00000223972"; transcript_id "ENST00000518655"; gene_name "DDX11L1"; gene_source "ensembl_havana"; gene_biotype "pseudogene"; transcript_name "DDX11L1-202"; transcript_source "ensembl";
chr1	transcribed_unprocessed_pseudogene	exon	11874	12227	.	+	.	gene_id "ENSG00000223972"; transcript_id "ENST00000518655"; exon_number "1"; gene_name "DDX11L1"; gene_source "ensembl_havana"; gene_biotype "pseudogene"; transcript_name "DDX11L1-202"; transcript_source "ensembl"; exon_id "ENSE00002269724";

**Comment by roryk** [comment]
Thanks, is this the ref-transcripts.gtf that comes with the hg19 installation in bcbio? If so I can test on that one.

**Comment by apastore** [comment]
Yes! I the original one. 

Thanks a lot!

**Comment by lbeltrame** [comment]
Looks like we hit this problem as well even in more recent days. We'll try by cleaning the cache.

**Comment by lbeltrame** [comment]
@roryk  I figured it out.

`make_pizzly_gtf` looks for features of type "gene", however, `hg19/rnaseq/ref-transcripts.gtf` does *not* contain any `gene` feature, only transcript. Therefore, the generated file is empty and everything breaks.

**Comment by lbeltrame** [comment]
```
cut -f3 hg38/rnaseq/ref-transcripts.gtf | grep gene | head
gene
gene
gene
gene
gene
gene
gene
gene
gene
gene

cut -f3 hg19/rnaseq/ref-transcripts.gtf | grep gene | head

```

**Comment by roryk** [comment]
Thanks-- yup, that is the problem. Sorry we haven't done much testing on hg19, we focused most of the work on that build on GRCh37. Could you run it with GRCh37 instead?

**Comment by lbeltrame** [comment]
That would mean a realignment of tens of samples done with hg19, so it's not really feasible. A proper solution would be to regenerate the GTF using the `prepare_tx_gff.py` script in cloudbiolinux. Simply rerunning it generated a proper GTF file with gene annotation and pizzly was able to run.

tl;dr - Regenerating the GTF files and offering them for download on upgrade would probably fix this.

**Comment by roryk** [comment]
Hi Luca,

Great-- I updated the hg19 transcriptome so it includes the 'gene' entries. I used the most recent Ensembl gene build I could find for GRCh37/hg19 which was 67, but I just now found where to find the current one: ftp://ftp.ensembl.org/pub/grch37/current/gtf/homo_sapiens/Homo_sapiens.GRCh37.87.gtf.gz, so I'll re-update with that.

**Comment by roryk** [comment]
Hi Luca,

Ok-- it is updated against 87 now, which is the most current version for GRCh37/hg19. Hopefully this fixes your issue. Thanks for figuring out what the issue is and sorry it took a while to get to.
